### PR TITLE
adding mmv2 functionalities

### PR DESCRIPTION
**Pull request by pallavi-pannu-oc**

### CODE REVIEW

File: dkube/sdk/api.py [hunk 1]
```diff
@@ -2443,7 +2443,7 @@ def modelmonitor_create(
             mm_config = super().get_modelmonitor_configuration(
                 response["uuid"])
             state = mm_config["status"]["state"]
-            if state.lower() in ["init", "ready", "error","incomplete"]:
+            if state.lower() in ["init", "ready", "error","pending"]:
```
Comment (songole) [inline dkube/sdk/api.py:2446]: include 'incomplete' also. If it is 'incomplete' they should update configuration until it reaches 'ready' state

File: dkube/sdk/api.py [hunk 2]
```diff
@@ -2813,7 +2813,7 @@ def modelmonitor_update(
         response = super().update_modelmonitor_config(id, config_dict)
         while wait_for_completion:
             mm_state = self.modelmonitor_get(id=id)["status"]["state"]
-            if mm_state.lower() in ["init", "error", "ready", "incomplete"]:
+            if mm_state.lower() in ["init", "error", "ready", "pending"]:
```
Comment (songole) [inline dkube/sdk/api.py:2816]: same as above

File: dkube/sdk/rsrcs/modelmonitor.py [hunk 3]
```diff
@@ -298,22 +298,217 @@ def __str__(self):
...
+        if version:
+            self.update_model_version(version)
+
+    def update_model_type(self, model_type=None):
+        """
+        Method to update the type of the model, check the Enum Class ModelType for the posible values
+        """
+        self.modelmonitor.model_type = model_type
+
+    def update_model_category(self, category=None):
```
Comment (songole) [inline dkube/sdk/rsrcs/modelmonitor.py:408]: we don't have this anymore

File: dkube/sdk/rsrcs/modelmonitor.py [hunk 4]
```diff
@@ -298,22 +298,217 @@ def __str__(self):
...
+        """
+        Method to update the category of the model, check the Enum Class ModelCategory for the possible values
+        """
+        self.modelmonitor.model_category = category
+
+    def update_model_version(self, version=None):
```
Comment (songole) [inline dkube/sdk/rsrcs/modelmonitor.py:414]: remove this as well

File: dkube/sdk/rsrcs/modelmonitor.py [hunk 5]
```diff
@@ -342,49 +537,25 @@ def update_drift_enabled(self, enabled=None):
         """
         Method to update the enable option in drift monitoring
         """
-        self.enabled = enabled
+        self.modelmonitor.drift_monitoring["enabled"] = enabled
```
Comment (songole) [inline dkube/sdk/rsrcs/modelmonitor.py:540]: A single update_drift_monitoring_details() with a few parameters would suffice. If any parameter is none, don't update that parameter.

update_drift_frequency(), update_drift_algorithm(), update_soft... etc are not required

File: dkube/sdk/rsrcs/modelmonitor.py [hunk 6]
```diff
@@ -420,60 +591,37 @@ def update_performance_enabled(self, enabled=None):
         """
         Method to update the enable option in performance monitoring
         """
-        self.enabled = enabled
+        self.modelmonitor.performance_monitoring["enabled"] = enabled
```
Comment (songole) [inline dkube/sdk/rsrcs/modelmonitor.py:594]: likewise, we just need 1 function which updates all performance monitoring details

File: dkube/sdk/rsrcs/modelmonitor.py [hunk 7]
```diff
@@ -511,37 +659,37 @@ def update_deployment_enabled(self, enabled=None):
         """
         Method to update the enable option in deployment monitoring
         """
-        self.enabled = enabled
+        self.modelmonitor.deployment_monitoring["enabled"] = enabled
```
Comment (songole) [inline dkube/sdk/rsrcs/modelmonitor.py:662]: likewise, a single function update_deployment_monitoring() is enough. We don't need a separate function for each field

File: dkube/sdk/rsrcs/modelmonitor.py [hunk 8]
```diff
@@ -135,9 +175,8 @@ class ModelCategory(Enum):
 
     """
 
-    AutoEncoder = "AutoEncoder"
-    TimeSeries = "TimeSeries"
-    Other = "Other"
+    TimeSeries = "time_series"
```
Comment (songole) [inline dkube/sdk/rsrcs/modelmonitor.py:178]: This function is not needed. category is gone

Comment (pallavi-pannu-oc) [reply dkube/sdk/rsrcs/modelmonitor.py]: @songole suggested changes are done in the latest commits. Please merge.

File: dkube/sdk/rsrcs/modelmonitor.py [hunk 9]
```diff
@@ -376,47 +357,22 @@ def update_basic(self, name, model_name, description, tags):
...
-        version=None,
     ):
         """
-        This function updates the DKube Modelmonitor configuration. The following updates are supported:
-            model type,
-            model category,
-            drift detection algorithm,
-            model version.
+        This function updates the DKube Modelmonitor configuration. The following update is supported:
+            model type
```
Comment (songole) [inline dkube/sdk/rsrcs/modelmonitor.py:363]: Can we merge this into update_basic(). This func is only updating model_type, right. The name would also be confusing.

In update_basic(), we don't have model, description and tags. Please remove.

**Pull request merged**